server.js

```diff
@@ -1,38 +1,155 @@
 
 // If you're using one of our datasets, uncomment the appropriate import below
 // to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
+
+// Database setup:
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+// Mongoose model setup:
+// Book Model
+const Book = mongoose.model("Book", {
+  bookID: {
+    type: Number,
+  },
+  title: {
+    type: String,
+  },
+  authors: {
+    type: String,
+  },
+  average_rating: {
+    type: Number,
+  },
+  isbn: {
+    type: String,
+  },
+  isbn13: {
+    type: String,
+  },
+  language_code: {
+    type: String,
+  },
+  num_pages: {
+    type: Number,
+  },
+  ratings_count: {
+    type: Number,
+  },
+  text_reviews_count: {
+    type: Number,
+  },
+});
+
+// -----------------------------------------------
+// To reset database and then populate db:
+// $ RESET_DATABASE=true npm run dev
+// Seed DATABASE using Async
+// forEach loop will put all Books from JSON into database
+if (process.env.RESET_DATABASE) {
+  console.log("Message: Resetting database");
+
+  const seedDatabase = async () => {
+    await Book.deleteMany();
+    await booksData.forEach((book) => new Book(book).save());
+  };
+  seedDatabase();
+}
+
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+// Middleware for handling if "no connection to Mongodb":
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next();
+  } else {
+    res.status(503).json({ error: "Service unavailable " });
+  }
+});
+
+// -----------------------------------------------
+// Start defining routes here:
+app.get("/", (req, res) => {
+  res.send(
+    "My API endpoints: /books ,/books/439554934 ,/sort?sort_by=average_rating ,/info/5eba83e3b3b1c00023249494  plus handling errors"
+  );
+});
+
+// To return all Books:
+// http://localhost:8080/books
+app.get("/books", async (req, res) => {
+  const books = await Book.find();
+  console.log(`Message: Found ${books.length} books`);
```

Comment:
I think this line was created for testing, so It is no need to leave it in the final project :)